### PR TITLE
adding support for the Nexys4DDR to PicoSoC example

### PR DESCRIPTION
**Pull request by cjearls**
Signed-off-by: Chandler Jearls <cjearls@vt.edu>

### CODE REVIEW

**Comment by mithro** [comment]
@kgugala

**Comment by kgugala** [comment]
the tests are running. I suppose you were waiting in a queue while the nightly build was running

**Comment by cjearls** [comment]
Awesome, looks like everything is passing

**Comment by cjearls** [comment]
It looks like the tests all passed, are we good to merge?

**Comment by cjearls** [comment]
@acomodi @rw1nkler Is there anything that needs to be done to get this merged?

File: xc7/picosoc_demo/README.rst [hunk 1]
```diff
@@ -65,6 +71,10 @@ The UART output should look as follows:
 
    PicoSoC uses baud rate of ``460800`` by default.
 
+.. note::
+
+   On the Nexys4DDR, the USB-UART does not work, so UART can be accessed from pins 1 and 2 of PMOD C.
```
Comment (mithro) [inline xc7/picosoc_demo/README.rst:76]: We should log an issue to figure out why the USB-UART is not working in the future.

File: xc7/picosoc_demo/README.rst [hunk 2]
```diff
@@ -65,6 +71,10 @@ The UART output should look as follows:
 
    PicoSoC uses baud rate of ``460800`` by default.
```
Comment (mithro) [inline xc7/picosoc_demo/README.rst:72]: This baud rate is kind of weird. @acomodi - Any idea why it is being used?

**Comment by mithro** [comment]
@kgugala - Will these changes be tested by CI?

File: xc7/picosoc_demo/README.rst (hunk 1 above)
Comment (cjearls) [reply xc7/picosoc_demo/README.rst:76]: My hypothesis is that it is related to the baud rate of 460800. I have used the USB-UART on this board successfully with projects that used a baud rate of 115200

Comment (cjearls) [reply xc7/picosoc_demo/README.rst:76]: I'm pretty sure that it is an issue related to the clock rate. I removed the BUFG that generates the clock and added a divider that slowed the entire design's clock by 4x, taking it from 100MHz to 25MHz. I don't know what clock rate the BUFG output is running at, but doing this allowed the onboard USB-UART to work at a baud rate of 57600

Comment (cjearls) [reply xc7/picosoc_demo/README.rst]: I fixed the problem, it is working with the USB-UART.

**Comment by kgugala** [comment]
Yes, this is run in CI

**Pull request merged**